pkg/list/lister.go

```diff
@@ -118,10 +121,16 @@ func clusterPackageShouldBeIncluded(item *result, options ListOptions) bool {
 
 func packageShouldBeIncluded(item *result, pkg *v1alpha1.Package, options ListOptions) bool {
 	if pkg != nil {
-		if !options.OnlyOutdated {
-			return true
+		if options.PackageName != "" && pkg.Spec.PackageInfo.Name != options.PackageName {
+			return false
 		}
-		return item.IndexItem != nil && pkg.Spec.PackageInfo.Version != item.IndexItem.LatestVersion
+		if options.OnlyOutdated && (item.IndexItem != nil && pkg.Spec.PackageInfo.Version == item.IndexItem.LatestVersion) {
+			return false
+		}
+		if options.Namespace != "" && pkg.Namespace != options.Namespace {
+			return false
+		}
```

Comment:
We can do this much earlier by adding the namespace to the packages request (see https://github.com/yeahservice/glasskube/blob/0e224f90f8e8e27cd23cabb13abedd3d3440cef9/pkg/list/lister.go#L184)